client/post-editor/media-modal/detail/index.jsx

```diff
@@ -74,10 +81,7 @@ class EditorMediaModalDetailBase extends React.Component {
 
 		return (
 			<div className="editor-media-modal-detail">
-				<HeaderCake
-					onClick={ onReturnToList }
-					backText={ this.props.translate( 'Media Library' ) }
-				/>
+				<HeaderCake onClick={ onReturnToList } backText={ this.getBackButton() } />
 				<DetailItem
```

Comment:
Instead of making a new method, we could use a ternary assignment directly in the props, similar to what you are doing in `client/my-sites/media/main.jsx` above.